core/src/tests/test_config_parser_dir.cc

```diff
@@ -29,8 +29,71 @@
 #include "lib/parse_conf.h"
 #include "dird/dird_globals.h"
 #include "dird/dird_conf.h"
+#include "lib/output_formatter_resource.h"
 
 namespace directordaemon {
+bool DoReloadConfig() { return false; }
+}  // namespace directordaemon
+
+namespace directordaemon {
+
+static std::string sprintoutput{};
+
+bool sprintit(void* ctx, const char* fmt, ...)
+{
+  va_list arg_ptr;
+  PoolMem msg;
+
+  va_start(arg_ptr, fmt);
+  msg.Bvsprintf(fmt, arg_ptr);
+  va_end(arg_ptr);
+
+  sprintoutput += msg.c_str();
+  return true;
+}
+
+TEST(ConfigParser_Dir, ParseSchedulerOddEvenDaysCorrectly)
+{
+  OSDependentInit();
+
+  std::string path_to_config_file = std::string(
+      RELATIVE_PROJECT_SOURCE_DIR "/configs/bareos-configparser-tests");
+  my_config = InitDirConfig(path_to_config_file.c_str(), M_ERROR_TERM);
+  my_config->ParseConfig();
+
+  ScheduleResource* even_schedule = nullptr;
+  ScheduleResource* odd_schedule = nullptr;
+
+  even_schedule = (directordaemon::ScheduleResource*)my_config->GetResWithName(
+      R_SCHEDULE, "Even Weeks");
+  odd_schedule = (directordaemon::ScheduleResource*)my_config->GetResWithName(
```

Comment:
```suggestion
ScheduleResource*  odd_schedule = (directordaemon::ScheduleResource*)my_config->GetResWithName(
```

Reply:
It is often the best way  to define and initialize a variable immediately before the first use.